Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,64 @@
-function setAlarm() {}
+let timer;
+let flashInterval;
+
+function setAlarm() {
+  const alarmInput = document.getElementById("alarmSet");
+  const timeRemaining = document.getElementById("timeRemaining");
+  let timeLeft = parseInt(alarmInput.value);
+
+  if (isNaN(timeLeft) || timeLeft <= 0) {
+    alert("Please enter a valid number of seconds.");
+    return;
+  }
+
+  updateTimerDisplay(timeLeft);
+
+    timer = setInterval(() => {
+    if (timeLeft > 0) {
+      timeLeft--;
+      updateTimerDisplay(timeLeft);
+    } else {
+      clearInterval(timer); 
+      timeRemaining.textContent = "Time Remaining: 00:00";
+      playAlarm(); 
+      flashBackground(); 
+    }
+  }, 1000);
+
+  document.getElementById("stop").addEventListener("click", () => {
+    clearInterval(timer);
+    pauseAlarm();
+    resetTimerDisplay();
+    stopFlashing();
+  });
+}
+
+
+function updateTimerDisplay(seconds) {
+  const minutes = Math.floor(seconds / 60);
+  const remainingSeconds = seconds % 60;
+  document.getElementById("timeRemaining").textContent =
+    `Time Remaining: ${String(minutes).padStart(2, '0')}:${String(remainingSeconds).padStart(2, '0')}`;
+}
+
+function resetTimerDisplay() {
+  document.getElementById("timeRemaining").textContent = "Time Remaining: 00:00";
```

Comment:
Imagine if you wanted to change this text from "Time Remaining" to "Time Left". Right now, you'd need to edit three different places in this file.

Can you think how to make it so you'd only need to make one change?